frontend/public/components/utils/async.tsx

```diff
@@ -67,12 +67,16 @@ export class AsyncComponent extends React.Component<AsyncComponentProps, AsyncCo
 
   render() {
     const { Component } = this.state;
+    const { LoadingComponent = LoadingBox } = this.props;
     const rest = _.omit(this.props, 'loader');
-    return Component != null ? <Component {...rest} /> : <LoadingBox />;
+    return Component != null ? <Component {...rest} /> : <LoadingComponent />;
   }
 }
 
-export type AsyncComponentProps = { loader: () => Promise<React.ComponentType> } & any;
+export type AsyncComponentProps = {
+  loader: () => Promise<React.ComponentType>;
+  LoadingComponent?: React.ReactNode;
+} & any;
```

Comment:
FYI, the `& any` basically broadens the type declaration to be assignable to anything.

```ts
type Foo = { x: string };
const f1: Foo = { x: 'foo' };
const f2: (Foo & any) = true; // breaks the structural typing contract
```

I realize it's part of the original code, but it looks fishy..

Reply:
We can improve that later, but I'd encourage not to use `type Foo = xxx & any` in new code, just like with `as any` type cast.